transactions/types/assets/create.py

```diff
@@ -27,15 +33,22 @@ def validate_create(self, tx_signers: list[str], recipients: list[tuple[list[str
         if not assets is None:
             if not isinstance(assets, list) and len(assets) != 1:
                 raise TypeError("`assets` must be a list of length 1 or None")
-            if "data" in assets[0] and not is_cid(assets[0]["data"]):
-                raise TypeError("`asset.data` must be a CID string")
+            if "data" in assets[0]:
+                if not is_cid(assets[0]["data"]) and assets[0]["data"] is not None:
```

Comment:
does is_cid process assets[0]["data"] = None without issues?
otherwise, it you could just swap the order of the if case:
if assets[0]["data"] is not None and not is_cid(assets[0]["data"])

Reply:
it shouldn't make a difference as is_cid(None) will return False but I reversed the order